Common/Securities/IRegisteredSecurityDataTypesProvider.cs

```diff
@@ -0,0 +1,44 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Collections.Generic;
+
+namespace QuantConnect.Securities
+{
+    /// <summary>
+    /// Provides the set of base data types registered in the algorithm
+    /// </summary>
+    public interface IRegisteredSecurityDataTypesProvider
```

Comment:
This guy is like a glorified `HashSet<Type>`, but I wanted to encapsulate the intent behind some explicit type names. We call `RegisterType` via `ISubscriptionDataConfigService.Add` (which is implemented via `DataManager.Add`). All subscription configs come through this point, so we take `config.Type` and register it with this provider. This gives us a _whitelist_ of types that the `DynamicSecurityData` implementation uses for determining whether or not to perform the empty list default value behavior.

We currently aren't called `UnregisterType`, but it felt wrong/incomplete to not include it.